rules/src/main/scala/fix/GuardrailScalaResponseTypes.scala

```diff
@@ -0,0 +1,46 @@
+package fix
+
+import java.util.Locale
+import scalafix.v1._
+import scala.meta._
+
+class GuardrailScalaResponseTypes extends SemanticRule("GuardrailScalaResponseTypes") {
+  private def cap(s: String): String = s.substring(0, 1).toUpperCase(Locale.US) + s.substring(1)
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    doc.tree.collect({
+      case defn @ Defn.Def(
+        _,
+        _,
+        _,
+        List(Term.Param(
+          Nil,
+          Term.Name("respond"),
+          Some(Type.Singleton(Term.Select(resourceType, Term.Name(responseType)))), None
+        )) :: otherArgs :: Nil,
+        Some(Type.Apply(
+          returnTypeName,
+          List(Type.Select(returnResourceType, Type.Name(returnResponseType)))
+        )),
+        _
+      ) =>
+        List(Patch.replaceTree(defn, defn.copy(
+          paramss = List(
+            List(Term.Param(
+              Nil,
+              Term.Name("respond"),
+              Some(Type.Singleton(Term.Select(
+                resourceType,
+                Term.Name(cap(responseType))
+              ))),
+              None
+            )),
+            otherArgs
+          ),
+          decltpe = Some(Type.Apply(
+            returnTypeName,
+            List(Type.Select(returnResourceType, Type.Name(cap(returnResponseType))))))
+        ).toString))
+    }).flatten.asPatch
```

Comment:
`flatten` here is just because you're returning a singleton list, but since it's never more or less than one element you can just strip out the unnecessary `List` and remove the `flatten`, just `collect` individual `Patch`es and `.asPatch` them together.